gemini/reasoning-engine/tutorial_graph_rag.ipynb

```diff
@@ -0,0 +1,1707 @@
HcCwXQwdG+3hPsdu/ezfXr1+v1V6vVGAwGPv74YwICAswBjVqtJjY2lvz8/AbzBQAcHBzYu3cvI0aMYPr06Zw4ccLqr8dWlJSUkJ+fz6BBg+q1abVabt68af6pK4K1ZMkSunTpwqEjGkzPhNIpYA7ec/6M0VCG9viOu/sfi8VkrMZ73ko6jZuN++hf4B2+CoWditvJW+6O4ds9GCtK8Z79JzoFzMFt1M/oNm8VNSW10w9lTZQujouLIykpicTERGJiYnjjjTdYu3Yt69atY//+/aSmplrrTyWEBQkGhFXNnDmTTZs2kZGRQWhoKBUVFY32HTZsGElJSaxZs4aEhATS0tLM2wHeeecdYmNjSUhIQK/Xm/cbO2oEzt16U3b+OPYePjh262f119HTw5nQ6dM5cuQI8+fPZ9OmTbz11lvmFRA/NWbMGFQqFRcuXLC46I8fP56srNolao0FAwAdOnTg4MGD+Pn5ERwczNmzZ63+mmxBXQDa0PLNwMBAvLy8zD+ffPIJP/zwAydPniQiIgKdydFcWdCha2+cej1H+aXaUtsmYw0Vuf/E+akx2HfqZj6mqmMXXAZMwHDtHEZDbcXB8svf4/iEPw7ed98ndh1ccRkYCMA1beOVCXfv3s3TTz+Nv7+/ReDy/PPPA7XJpkK0BQkGhNXNnz+f6Ohojh49SlhYGNXVDX8TWrNmDcOGDWP58uXMmTOHDRs2ADBixAjef/99Tp06RUREBHPmzKk37TA2aCYALoOsf1fATqlgYv+uREZGsnjxYhITE3n77bfJyMjgq6++anDVhIuLC0OGDAEskwTrAgAfH58GM9d/zM3NjcTERLp168aUKVPIzs624quyDa6utcl5DRVs2rhxI4cOHWL79u3mbVeuXAFqH0ZUWW206G/v4YOxvARjZQXGshJMVQbsuzxR77j2Hj5gMlJdUvserS4pQtW5gVyFf2+rqm58pcLFixfJzMy0CFq8vLzo378/AEVFRU2+fiHul6Qui/sWERFBREREg22LFy9m8eLF5t+joqKIioqy6OPn58fRo0cb3H/58uUsX7680XOP6teVJBS4DAhs8Xi7zf2w0TZVJ298lx0EajO/543uiaOjI9HR0URHR1v0bWxev6Hb+0888US9qYY6W7duZevWrRbbPDw8yMy07hp3W+Lu7k737t0bvLNSl0PQWAKfg+rBfDeyVzWeE2M0Ghk8eDBr1qxpsL2tl+8K2yXBgHjsmEwmDuzagbffEBw7e99zqdim2CkVjO3jIcu/HmMhISF8+umnnDhxgpEjRzbZt+5uzYULF5g11wUFmKcKqoqvoezghtLBCZPKHoW9I1XF9fNFqoqvgUKJyq12pYjKrSvVt3+o169u25OdGi881LdvX06dOsWkSZPa7JHfQjREpgnEY0Ov1xMfH8+CBQs4c+YMf353GSqldT8wVUoFH8wcbNVjigdr6dKlODs789prr1FYWFiv/cd3arp3785zzz3Htm3bqCovpee/KwRW3silIuefdOg7HACF0g6nXkMou/itxQqDGv1t9OeO4vjkAJSOtft26D0Uw/XzVBZevtuvXIc+8wgAzo6Nfwf75S9/yfXr19m0aVO9tvLycovcGSGsSe4MiMfGjRs3ePnll+nUqRORkZG8PjcMl/Q8lu213lMG/xg6UErGPuaeeuop4uLimDNnDn5+fuYKhCaTiZycHOLi4lAqlTz55JMArFq1iuDgYMaMGYPP6BBKLhdy57u/onR0xj3gZfNxO40PpyL3JAU7luI6JASUSkpPfo2puorOE++WmXYb/XP0mckUJizHddh0lPaO6E5/g8rdi8oKXZPf+MPDw9m1axcLFy4kOTmZcePGUVNTw/nz59m1axeJiYkMHz687f54wmYpTI1NaP5ISUkJ7u7u3LlzBzc3twcxLiFabF3yRas8VGbJVD/+Y6L1VyaIh+PSpUusXr2aQ4cOce3aNRQKBb6+vgQGBrJw4ULzEzABDh8+zIoVK/g+IwODUYGTzyA6B87H3tNyjr6y4BK3j27DcP1fYDLi0N2PzhPCcXziact+hZcpPrQRww9Z2Dm74zo0BIW9E7eTNlJQUIC3tzfQ8IOKqqqqiImJ4fPPPyc7OxtnZ2f69OlDaGgov/3tb+UzWNyTll6/JRgQ7UJCeh4rDmRSbTTdUw6BnVKBSqngj6EDG3yqnLA94ZvTSL18y+q5KA4nPif3+H5KS0uxs7N+kSwhGtLS67fkDIh2YfaIniT9bgJj+9Q+M8CumVyCuvaxfTxI+t0ECQSE2QczB7c6F8VYZVk6WFGho+D7QwQEBEggIB5JcmdAtDsXC3XsSMsjOauIvFtl/PgNrqC2oNDE/l2ZN7qnrBoQDUpoZS5K/mdv4tRzMPYePtTotdhnH0F7s5DDhw83WJZaiLYi0wRCAHpDNbm39FRWG3FQKenl4YJLE9ncQtRpTS7K7aPbKDufQo3uFvYqJSOHD2PFihVMnjzZyqMUomkSDAghRCtJLop43EnOgBBCtJLkoghbIfdLhRCiCT5dnIl9fZTkooh2TaYJhBDiHkkuinhctPT6Le9eIYS4Ry6OKgb2cH/YwxDCaiRnQAghhLBxEgwIIYQQNk6CASGEEMLGSTAghBBC2DgJBoQQQggbJ8GAEEIIYeMkGBBCCCFsnAQDQgghhI2TYEAIIYSwcRIMCCGEEDZOggEhhBDCxkkwIIQQQtg4CQaEEEIIGyfBgBBCCGHjJBgQQgghbJwEA0IIIYSNU7Wkk8lkAqCkpKRNByOEEEII66m7btddxxvTomBAp9MB4OPj08phCSGEEOJB0+l0uLu7N9quMDUXLgBGo5H8/HxcXV1RKBRWHaAQQggh2obJZEKn09GjRw+UysYzA1oUDAghhBCi/ZIEQiGEEMLGSTAghBBC2DgJBoQQQggbJ8GAEEIIYeMkGBBCCCFsnAQDQgghhI2TYEAIIYSwcf8P02jE8qUnC0UAAAAASUVORK5CYII=",
```

Comment:
Remove these image outputs, they make the notebook very large and difficult to load.